src/sign/participant.rs

```diff
@@ -156,39 +158,69 @@ impl ProtocolParticipant for SignParticipant {
     where
         Self: Sized,
     {
-        todo!()
+        let config = ParticipantConfig::new(id, &other_participant_ids)?;
+        Ok(Self {
+            sid,
+            config,
+            input,
+            storage: Default::default(),
+            status: Status::NotReady,
+        })
     }
 
     fn id(&self) -> ParticipantIdentifier {
-        todo!()
+        self.config.id()
     }
 
     fn other_ids(&self) -> &[ParticipantIdentifier] {
-        todo!()
+        self.config.other_ids()
     }
 
     fn process_message<R: RngCore + CryptoRng>(
         &mut self,
         rng: &mut R,
         message: &Message,
```

Comment:
This is beyond the scope of this PR but I wonder if we could introduce, _just one more associated type_ for `ProtocolParticipant` which specifies the `MessageType` then `process_message` would take a `message: Self::MessageType` instead of the generic `Message` type.

Reply:
Ooh, interesting. I thing we would want to revisit the `Message` type more broadly; there are definitely ways it could be improved. Currently, `MessageType` is a field on `Message`, but it doesn't contain any actual content so we couldn't pass only that to this method. It's just an indicator as to what the message contains. The `Message` also contains `unverified_bytes`, which you have to manually convert to the type you expect it to be (and that type is not encoded anywhere except in ad-hoc `from_message` methods around the code).

We've also talked about generalizing `Message`s to encode how they are supposed to be sent over the network, rather than e.g. hiding away the fact of broadcast in the library.